pkg/operator/controllers/banner/banner_controller_test.go

```diff
@@ -0,0 +1,182 @@
+			featureFlag:     false,
+		},
+		{
+			name: "Banner not modified when feature disabled",
+			oldCN: consolev1.ConsoleNotification{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: BannerName,
+				},
+				Spec: consolev1.ConsoleNotificationSpec{
+					Text:            "OLD BANNER TEXT",
+					Location:        consolev1.BannerTop,
+					Color:           "#000",
+					BackgroundColor: "#ff0",
+				},
+			},
+			bannerSetting:   "ContactSupport",
+			expectBanner:    true,
+			expectedMessage: "OLD BANNER TEXT",
+			featureFlag:     false,
+		},
+		{
+			name:          "No banner from 0 ConsoleNotifications",
+			bannerSetting: "",
+			expectBanner:  false,
+			featureFlag:   true,
+		},
+		{
+			name:            "Support banner from 0 ConsoleNotifications",
+			bannerSetting:   "ContactSupport",
+			expectBanner:    true,
+			expectedMessage: "We have noticed an issue regarding your cluster requiring an action on your part. Please contact support with your cluster resource ID: FAKE_RESOURCE_ID",
+			featureFlag:     true,
+		},
+		{
+			name: "Delete existing ConsoleNotification",
+			oldCN: consolev1.ConsoleNotification{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: BannerName,
+				},
+				Spec: consolev1.ConsoleNotificationSpec{
+					Text:            "OLD BANNER TEXT",
+					Location:        consolev1.BannerTop,
+					Color:           "#000",
+					BackgroundColor: "#ff0",
+				},
+			},
+			bannerSetting: "",
+			expectBanner:  false,
+			featureFlag:   true,
+		},
+		{
+			name: "Support banner from existing ConsoleNotification",
+			oldCN: consolev1.ConsoleNotification{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: BannerName,
+				},
+				Spec: consolev1.ConsoleNotificationSpec{
+					Text:            "OLD BANNER TEXT",
+					Location:        consolev1.BannerTop,
+					Color:           "#000",
+					BackgroundColor: "#ff0",
+				},
+			},
+			bannerSetting:   "ContactSupport",
+			expectBanner:    true,
+			expectedMessage: "We have noticed an issue regarding your cluster requiring an action on your part. Please contact support with your cluster resource ID: FAKE_RESOURCE_ID",
+			featureFlag:     true,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			instance := arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "cluster",
+				},
+				Spec: arov1alpha1.ClusterSpec{
+					ResourceID: "FAKE_RESOURCE_ID",
+					Banner: arov1alpha1.Banner{
+						Content: arov1alpha1.BannerContent(tt.bannerSetting),
+					},
+					Features: arov1alpha1.FeaturesSpec{
+						ReconcileBanner: tt.featureFlag,
+					},
+				},
+			}
+
+			r.arocli = arofake.NewSimpleClientset(&instance)
+			r.consolecli = consolefake.NewSimpleClientset(&tt.oldCN)
+
+			// function under test
+			_, err := r.Reconcile(context.Background(), ctrl.Request{})
+
+			if err != nil {
+				if tt.expectedErr == nil {
+					t.Error(err)
+				}
+				if !strings.EqualFold(err.Error(), tt.expectedErr.Error()) {
+					t.Error(err)
+				}
+				return
+			}
```

Comment:
We need to test that:

1. When the unit under test return an error, we need to make sure that this is the error we expet (you handled this)
2. When there is no error we need to check that this is what we expect (you do not check it)

Below is the pattern which we often use (`wantErr` is a `string`).

```suggestion
			if err != nil && err.Error() != tt.wantErr ||
				err == nil && tt.wantErr != "" {
				t.Error(err)
			}
```